### PR TITLE
Addition of constructor

### PR DESCRIPTION
**Pull request by matthewhegarty**
Added constructor to prevent default read of Group table.
See [issue #6](https://github.com/computer-lab/django-rest-framework-roles/issues/6)

### CODE REVIEW

File: drf_roles/mixins.py [hunk 1]
```diff
@@ -19,7 +18,14 @@ class RoleError(Exception):
 class RoleViewSetMixin(object):
     """A ViewSet mixin that parameterizes DRF methods over roles"""
     _viewset_method_registry = set(getattr(settings, "VIEWSET_METHOD_REGISTRY", DEFAULT_REGISTRY))
-    _role_groups = set(getattr(settings, "ROLE_GROUPS", DEFAULT_GROUPS))
+
+    def __init__(self, **_kwargs):
```
Comment (r1b) [inline drf_roles/mixins.py:23]: Why `**_kwargs`?

File: drf_roles/mixins.py [hunk 2]
```diff
@@ -19,7 +18,14 @@ class RoleError(Exception):
...
+        groups = []
+        if hasattr(settings, "ROLE_GROUPS"):
+            groups = set(getattr(settings, "ROLE_GROUPS", groups))
```
Comment (r1b) [inline drf_roles/mixins.py]: You can just convert to a set once at the end.

File: drf_roles/mixins.py [hunk 3]
```diff
@@ -19,7 +18,14 @@ class RoleError(Exception):
 class RoleViewSetMixin(object):
     """A ViewSet mixin that parameterizes DRF methods over roles"""
     _viewset_method_registry = set(getattr(settings, "VIEWSET_METHOD_REGISTRY", DEFAULT_REGISTRY))
-    _role_groups = set(getattr(settings, "ROLE_GROUPS", DEFAULT_GROUPS))
+
+    def __init__(self, **_kwargs):
+        groups = []
+        if hasattr(settings, "ROLE_GROUPS"):
```
Comment (r1b) [inline drf_roles/mixins.py]: Don't need this guard if you are providing a default to `getattr`.

File: drf_roles/mixins.py [hunk 4]
```diff
@@ -19,7 +18,14 @@ class RoleError(Exception):
...
+            groups = set(getattr(settings, "ROLE_GROUPS", groups))
+        else:
+            groups = [group.name.lower() for group in Group.objects.all()]
+        self._role_groups = set(groups)
```
Comment (r1b) [inline drf_roles/mixins.py:27]: I would just inline `groups` here like:

```python
self._role_groups = set(getattr(settings, "ROLE_GROUPS", [group.name.lower() for group in Group.objects.all()]))
```

**Review by r1b — CHANGES_REQUESTED** [review]
Otherwise this is the right idea, thanks!

File: drf_roles/mixins.py (hunk 1 above)
Comment (matthewhegarty) [reply drf_roles/mixins.py:23]: It is the [`initkwargs`](https://github.com/encode/django-rest-framework/blob/master/rest_framework/viewsets.py#L96) param passed when constructing the `ViewSet`.  If this is not included then the process crashes with:

```
  File "./lib/python3.6/site-packages/rest_framework/viewsets.py", line 96, in view
    self = cls(**initkwargs)
TypeError: __init__() got an unexpected keyword argument 'suffix'
```

File: drf_roles/mixins.py (hunk 4 above)
Comment (matthewhegarty) [reply drf_roles/mixins.py]: Good call: 24c54fee625577301d2205c107fcfabdabaf246e

Comment (matthewhegarty) [reply drf_roles/mixins.py:27]: The side-effect of this is that there is always a read on the `auth.group` table.  The original approach avoids that extra hit.

File: drf_roles/mixins.py (hunk 2 above)
Comment (matthewhegarty) [reply drf_roles/mixins.py]: :+1:  a62647fe

File: drf_roles/mixins.py (hunk 3 above)
Comment (matthewhegarty) [reply drf_roles/mixins.py]: 7fe82502

File: drf_roles/mixins.py (hunk 4 above)
Comment (r1b) [reply drf_roles/mixins.py:27]: Ah I see, because the arguments to getattr are evaluated 👍

**Review by r1b — APPROVED** [review]
Looks good!

**Comment by r1b** [comment]
Thanks again, I'll cut a release to pypi this weekend.

**Comment by r1b** [comment]
Closes #6

**Comment by matthewhegarty** [comment]
Please can you update here when the release is available? thanks

**Comment by r1b** [comment]
@matthewhegarty sorry for the delay - it's live now

**Comment by matthewhegarty** [comment]
@r1b :+1: thanks

**Pull request merged**